core/src/code/utils.rs

```diff
@@ -160,37 +160,64 @@ pub fn check_imports(module: &Module) -> Result<(), CodeError> {
 
     let syscalls = SyscallName::instrumentable_map();
 
+    const EXPECTED_NUMBER_OF_MEMORY_IMPORTS: usize = 1;
+    let mut number_of_memory_imports = 0;
     let mut visited_imports = BTreeSet::new();
-    for (import_index, import) in imports.iter().enumerate() {
-        let External::Function(i) = import.external() else {
-            continue;
-        };
-
-        // Panic is impossible, unless the Module structure is invalid.
-        let Type::Function(func_type) = &types
-            .get(*i as usize)
-            .unwrap_or_else(|| unreachable!("Module structure is invalid"));
-
-        let syscall = syscalls
-            .get(import.field())
-            .ok_or(ImportError::UnknownImport(import_index as u32))?;
 
-        if !visited_imports.insert(*syscall) {
-            Err(ImportError::DuplicateImport(import_index as u32))?;
+    for (import_index, import) in imports.iter().enumerate() {
+        let import_index: u32 = import_index
+            .try_into()
+            .unwrap_or_else(|_| unreachable!("Import index should fit in u32"));
+
+        match import.external() {
```

Comment:
- add new tests to `core/src/code/mod.rs` for cases when a global/table/2 memories is imported
- remove https://github.com/gear-tech/gear/blob/8ba3c6996be2c4fd060f2d9f16b292c72b201f2c/pallets/gear/src/tests.rs#L13055-L13057